learning_assistant/api.py

```diff
@@ -23,3 +32,69 @@ def get_setup_messages(course_id):
         setup_messages = [{'role': 'system', 'content': x} for x in message_content]
         return setup_messages
     return None
+
+
+def _extract_block_contents(child, category):
+    """
+    Process the child contents based on its category.
+
+    Returns a string or None if there are no contents available.
+    """
+    if category == 'html':
+        content_html = child.get_html()
+        text = html_to_text(content_html)
+        return text
+
+    if category == 'video':
+        transcript = get_text_transcript(child)  # may be None
+        return transcript
+
+    return None
+
+
+def _leaf_filter(block):
```

Comment:
This is a custom leaf filter to exclude blocks that are not video or html, that way we limit any calculations/processing on nodes that aren't relevant to what we need.